src/Controller/Framework/AwsStorageController.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace App\Controller\Framework;
+
+use App\Command\CommandDispatcherTrait;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+use League\Flysystem\AwsS3v3\AwsS3Adapter;
+use League\Flysystem\Filesystem;
+use Symfony\Component\HttpFoundation\Request;
+use Aws\Credentials\CredentialProvider;
+use App\Command\Framework\AddFileToAwsCommand;
+use App\Command\Framework\DeleteFileCommand;
+use App\Entity\Framework\LsItem;
+use Symfony\Component\HttpFoundation\JsonResponse;
+
+class AwsStorageController extends AbstractController
+{
+    use CommandDispatcherTrait;
+
+    /**
+     * @Route("/cfdoc/{id}/{field}/aws", name="aws_storage_file")
+     */
+    public function awsStorage(Request $request, LsItem $lsItem, string $field)
+    {
+        $output = ['uploaded' => false];
+        $filesystem = $this->configuration();
+        $file = $request->files->get('file');
+
+        $fileName = $file->getClientOriginalName();
+        $name = explode('.', $fileName);
+        $fileName = $name[0].'-'.md5(uniqid()).'.'.$file->getClientOriginalExtension();
+        $fileName = preg_replace('/[^A-Za-z0-9\.]/', '_', $fileName);
+        $filePath = $file->getRealPath();
+        if (!$filesystem->has($fileName)) {
+            $stream = fopen($filePath, 'r+');
+            fclose($stream);
+        }
+
+        $command = new AddFileToAwsCommand($lsItem, $fileName, $field);
+        $this->sendCommand($command);
+
+        $output['uploaded'] = true;
+        $output['fileName'] = $fileName;
+
+        return new JsonResponse($output);
+    }
+
+    private function configuration()
+    {
+        $provider = CredentialProvider::defaultProvider();
+        $client = \Aws\S3\S3Client::factory([
+    /* 'credentials' => [
+      'key'    => 'AKIAJMM3WLA2KVT732XA',
+      'secret' => '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'
+    ], */
+    'credentials' => $provider,
+    'region' => 'us-east-1',
+    //'region'  => 'eu-central-1',
+    'version' => 'latest',
+    ]);
+        //$adapter    = new AwsS3Adapter($client, "sample-test4");
+        $adapter = new AwsS3Adapter($client, 'actinc.opensalt.np', 'dev');
+        $filesystem = new Filesystem($adapter);
+
+        return $filesystem;
+    }
+
+    /**
+     * @param string $fileName
+     *
+     * @Route("/{fileName}/file-download", requirements={"fileName"=".+"}, name="aws_file_download")
```

Comment:
We should use a subdirectory here instead of allowing `{fileName}` to be in the top directory.